internal/cf/client.go

```diff
@@ -51,8 +56,10 @@ type clientCacheEntry struct {
 }
 
 var (
-	cacheMutex  = &sync.Mutex{}
-	clientCache = make(map[clientIdentifier]*clientCacheEntry)
+	cacheMutex        = &sync.Mutex{}
```

Comment:
With your changes, we now have to caches.
To make it clearer, use more specific terms like "clientCacheMutex" also for this variable